### PR TITLE
Release 1.7.1-gcm.0

### PR DESCRIPTION
**Pull request by wallrj**
* [Draft Release](https://github.com/jetstack/jetstack-secure-gcm/releases/tag/untagged-ca86324b5d293158d6c6) 
* [Submitted for review](https://console.cloud.google.com/partner/editor/jetstack-public/jetstack-secure-for-cert-manager?project=jetstack-public) 

|                 |                                                                                                               |
| --------------- | ------------------------------------------------------------------------------------------------------------- |
| `README.md`     | https://github.com/jetstack/jetstack-secure-gcm/blob/release-1.7/README.md                                    |
| deployer image  | gcr.io/jetstack-public/jetstack-secure-for-cert-manager/deployer:1.7.1-gcm.0                                  |
| deployer digest | 4ad6ad2262e4                                                                                                  |
| git tag         | 1.7.0-gcm.0                                                                                                   |
| git commit      | 3ccd5b15285c5cf55989a3f68340b09b2ffd9166                                                                      |
| retag logs      | https://console.cloud.google.com/cloud-build/builds/59ffed2d-9773-4f19-be76-c228197e9a20?project=885059085598 |
| deployer logs   | https://console.cloud.google.com/cloud-build/builds/e16dbf32-b48b-4153-8c86-91040b9d822c?project=885059085598 |

### CODE REVIEW

**Comment by wallrj** [comment]
@maelvls The smoke-test failed, so I'll probably need your help diagnosing that. The links to the logs are in the PR description.

File: chart/jetstack-secure-gcm/charts/cert-manager/templates/rbac.yaml [hunk 1]
```diff
@@ -133,7 +133,7 @@ rules:
     verbs: ["create", "delete", "get", "list", "watch"]
   - apiGroups: [""]
     resources: ["secrets"]
-    verbs: ["get", "list", "watch", "create", "update", "delete"]
+    verbs: ["get", "list", "watch", "create", "update", "delete", "patch"]
```
Comment (wallrj) [inline chart/jetstack-secure-gcm/charts/cert-manager/templates/rbac.yaml:136]: This new verb was added for the server-side apply features in 1.7

File: chart/jetstack-secure-gcm/charts/cert-manager/templates/webhook-config.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  namespace: {{ .Release.Namespace | quote }}
+  labels:
+    app.kubernetes.io/name: {{ include "cert-manager.name" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/component: "webhook"
+data:
+  {{- if .Values.webhook.config }}
+  config.yaml: |
+    {{ .Values.webhook.config | toYaml | nindent 4 }}
+  {{- end }}
```
Comment (wallrj) [inline chart/jetstack-secure-gcm/charts/cert-manager/templates/webhook-config.yaml:23]: This is the new config file thing in 1.7

File: cloudbuild.yaml [hunk 3]
```diff
@@ -189,6 +189,31 @@ steps:
...
+            --namespace "$ns" \
+            --follow \
+            --all-containers \
+            --prefix \
+            --selector app.kubernetes.io/name=jetstack-secure-gcm || true
+          sleep 10
+        done
+    waitFor:
+      - check-cloud-marketplace-tools
+      - push-deployer
```
Comment (wallrj) [inline cloudbuild.yaml:215]: This is a new step which collects the logs of all the JSP-GCM components for easier debugging when the smoke-tests fail.
It gets the logs of all the containers of all the pods having label `app.kubernetes.io/name=jetstack-secure-gcm` and prefixes each log line with the ns/pod/container

This helped me figure out that I'd forgotten to add the missing `Secret > Patch` RBAC to the Schema.yaml file.

**Review by wallrj — COMMENTED** [review]
@maelvls I've submitted this for review, but @james-w expects it to be rejected because in this release (1.7) we removed old API versions but Google's test framework uses one version of the cert-manager CRDs.

I'll await an email from Dinesh.

**Review by wallrj — COMMENTED** [review]
I did this change as a separate commit so that it's easier to apply the same changes in future.
I added the following new step to the [release-process](https://github.com/jetstack/jetstack-secure-gcm/releases/tag/untagged-ca86324b5d293158d6c6):

```
[ ] Apply the ubbagent patch.

This adds a sidecar container called ubbagent (Universal Billing Agent) to the main cert-manager deployment.

patch -p1 < <(curl -sSL https://github.com/jetstack/jetstack-secure-gcm/commit/e1ecea1cb623358d3181157fcf9bff04fd41988f.diff)
```

File: cloudbuild.yaml (hunk 3 above)
Comment (wallrj) [reply cloudbuild.yaml:215]: Here's what the output looks like: https://console.cloud.google.com/cloud-build/builds/e16dbf32-b48b-4153-8c86-91040b9d822c;step=13?project=885059085598

Comment (maelvls) [reply cloudbuild.yaml:215]: Thanks, very useful indeed!

**Comment by maelvls** [comment]
Excellent work thank you for adding the new cloudbuild task.

I propose that we copy in a separate PR the "release process" from your GitHub release description to a `docs/release-process.md` page.

**Review by maelvls — APPROVED** [review]

**Pull request merged**